src/main/java/edu/kit/ipd/crowdcontrol/objectservice/notification/SQLEmailNotificationPolicy.java

```diff
@@ -0,0 +1,116 @@
+package edu.kit.ipd.crowdcontrol.objectservice.notification;
+
+import edu.kit.ipd.crowdcontrol.objectservice.database.model.tables.records.NotificationTokenRecord;
+import edu.kit.ipd.crowdcontrol.objectservice.database.operations.NotificationOperations;
+import edu.kit.ipd.crowdcontrol.objectservice.mail.MailSender;
+import edu.kit.ipd.crowdcontrol.objectservice.template.Template;
+import org.jooq.Field;
+import org.jooq.Record;
+import org.jooq.Result;
+import org.jooq.impl.DSL;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * The SQLEmailNotificationPolicy checks given queries in a SQL-Database and sends notifications via email.
+ *
+ * @author Simon Korz
+ * @version 1.0
+ */
+public class SQLEmailNotificationPolicy extends NotificationPolicy<List<String>> {
+    private MailSender mailSender;
+    private NotificationOperations operations;
+
+
+    /**
+     * @param mailSender an implementation of the MailSender interface
+     * @param operations  instance of the notification operations
+     */
+    public SQLEmailNotificationPolicy(MailSender mailSender, NotificationOperations operations) {
+        super(operations);
+        this.mailSender = mailSender;
+        this.operations = operations;
+    }
+
+    private boolean resultHasIdAndTokenField(Result<Record> result) {
+        Field<Integer> idField = DSL.field("id", Integer.class);
+        Field<String> tokenField = DSL.field("token", String.class);
+
+        return (result.get(0).field(idField) != null) && (result.get(0).field(tokenField) != null);
+    }
+
+    @Override
+    protected List<String> check(Notification notification) {
+        Result<Record> result = operations.runReadOnlySQL(notification.getQuery());
```

Comment:
its not a deal breaker....i just wondered why you optimised the cheaper method ;)